agent/connect/ca/provider_vault.go

```diff
@@ -232,11 +257,12 @@ func (v *VaultProvider) renewToken(ctx context.Context, watcher *vaultapi.Lifeti
 					continue
 				}
 			}
-
 			// Restart the watcher.
+
 			go watcher.Start()
 
 		case <-watcher.RenewCh():
+			retrier.Reset()
```

Comment:
Does this get called after every successful request to vault from the watcher? For example does it fire after the initial token fetch/validation? If not, we might need to also reset above when we re-start the watcher?

Reply:
This gets called on every successful renewal which should happen immediately after start and then the LifetimeWatcher sleeps.